src/Effect/ReadFS.hs

```diff
@@ -114,6 +119,13 @@ readContentsBS' path = send (ReadContentsBS' path)
 readContentsBS :: (Has ReadFS sig m, Has Diagnostics sig m) => Path b File -> m ByteString
 readContentsBS = fromEither <=< readContentsBS'
 
+fileIsBinary :: (Has ReadFS sig m, Has Diagnostics sig m) => Path b File -> m Bool
+fileIsBinary file = do
+  -- git defines binaries as "a file that contains a 0 byte in the first 8000 bytes of the file".
+  attemptedContent <- send (ReadContentsBSLimit' file 8000)
+  content <- fromEither attemptedContent
+  pure $ BS.elem 0 content
```

Comment:
A few suggested changes:

- move `fileIsBinary` into `App.Fossa.BinaryDeps`
- move the git comment to a function-level `-- | Doc string` on `fileIsBinary`
- export the `readContentsBSLimit` operation from `Effect.ReadFS`, similar to the existing `readContentsBS` operation:

```hs
-- name clashes with an existing function you defined; might have to rename it
readContentsBSLimit' :: Has ReadFS sig m => Path b File -> Int -> m (Either ReadFSErr ByteString)
readContentsBSLimit' file limit = send (ReadContentsBSLimit' file limit)

readContentsBSLimit :: (Has ReadFS sig m, Has Diagnostics sig m) => Path b File -> Int -> m ByteString
readContentsBSLimit file limit = readContentsBSLimit' file limit >>= fromEither
```

Reply:
I'm OK to move `fileIsBinary` to `BinaryDeps`. I put it here because at least in monorepo land we've found it super useful to be able to avoid inspecting files for eg IPR data when they're binary, so I thought it may be similarly generally useful here, but definitely happy to only make it more general when/if that need actually shows up.